### DISCUSSION TITLE
How do I find the Y level of terrain at a specific position?

### DISCUSSION THREAD

**Discussion by vismodo** [pokepetter/ursina#284] [Q&A] [answered]
Hi! I'm trying to generate trees in my game. However, I'm using procedurally generated terrain and I need to place trees at different Y levels. How do I find the Y level of terrain at a specific position using using Ursina? I found a function called 'get_y()' but I don't know how to use it. Thanks :)

**Comment by leer-ursina** [comment, marked as answer]
hello,
what about random x,y positions in a "for xy in range( ):"
with raycast up at this position
then get the hit point of raycast vs terrain

your specific_position should be a vec3 type => vec3( x, y, z )

y = specific_position.get_y()
or specicific_position[y]

**Comment by vismodo** [comment]
Thanks! I figured it out.
